src/Games/NexusMods.Games.StardewValley/Installers/SMAPIInstaller.cs

```diff
@@ -79,71 +101,78 @@ private SMAPIInstaller(IOSInformation osInformation, IFileHashCache fileHashCach
+        {
+            downloadId = await _fileOriginRegistry.RegisterDownload(
+                installDataFile.Item.StreamFactory!,
+                new FilePathMetadata { OriginalName = installDataFile.Item.FileName, Quality = Quality.Low },
+                cancellationToken
+            );
+        }
         else
         {
-            downloadId = found.First();
+            downloadId = found[0];
         }
 
         var gameFolderPath = info.Locations[LocationId.Game];
         var archiveContents = (await _fileOriginRegistry.Get(downloadId)).GetFileTree();
 
-        // TODO: install.dat is an archive inside an archive see https://github.com/Nexus-Mods/NexusMods.App/issues/244
-        // the basicFiles have to be extracted from the nested archive and put inside the game folder
-        // https://github.com/Pathoschild/SMAPI/blob/9763bc7484e29cbc9e7f37c61121d794e6720e75/src/SMAPI.Installer/InteractiveInstaller.cs#L380-L384
-        /*
-        var basicFiles = archiveContents.EnumerateChildrenBfs()
-            .Where(kv => !kv.Value.Item.Path.Equals("unix-launcher.sh")).ToArray();
+        // see original installer:
+        // https://github.com/Pathoschild/SMAPI/blob/5919337236650c6a0d7755863d35b2923a94775c/src/SMAPI.Installer/InteractiveInstaller.cs#L384
 
-        if (_osInformation.IsLinux || _osInformation.IsOSX)
-        {
-            // TODO: Replace game launcher (StardewValley) with unix-launcher.sh by overwriting the game file
-            // https://github.com/Pathoschild/SMAPI/blob/9763bc7484e29cbc9e7f37c61121d794e6720e75/src/SMAPI.Installer/InteractiveInstaller.cs#L386-L417
-            var modLauncherScriptFile = archiveContents.EnumerateChildrenBfs()
-                .First(kv => kv.Value.Item.Segment.Equals("unix-launcher.sh"));
+        var isUnix = _osInformation.IsUnix();
+        var isXboxGamePass = info.Store == GameStore.XboxGamePass;
 
-            var gameLauncherScriptFilePath = gameFolderPath.Combine("StardewValley");
+        // add all files from the archive
+        foreach (var kv in archiveContents.GetFiles())
+        {
+            var to = new GamePath(LocationId.Game, kv.Path());
+
+            // For Linux & macOS: replace the game launcher executable "StardewValley" with "unix-launcher.sh"
+            // https://github.com/Pathoschild/SMAPI/blob/5919337236650c6a0d7755863d35b2923a94775c/src/SMAPI.Installer/InteractiveInstaller.cs#L395-L425
+            if (isUnix && kv.Item.FileName.Equals("unix-launcher.sh"))
+            {
+                to = new GamePath(LocationId.Game, "StardewValley");
+            }
+
+            // For Xbox Game Pass: replace "Stardew Valley.exe" with "StardewModdingAPI.exe"
+            // https://stardewvalleywiki.com/Modding:Installing_SMAPI_on_Windows#Xbox_app
+            if (isXboxGamePass && kv.Item.FileName.Equals("StardewModdingAPI.exe"))
+            {
+                to = new GamePath(LocationId.Game, "Stardew Valley.exe");
+            }
```

Comment:
So "Stardew Valley.exe" is some kind of launcher that isn't needed for executing the game?

Reply:
I'm not sure on the specifics, especially with the Xbox version.

Reply:
I think the executable doesn't do much, the main code is in the `Stardew Valley.dll` (it's a C# game).

Reply:
No chance that SMAPI relies on the backup of that launcher file being where its install script puts it?